diff --git a/lib/useEnsuredForwardedRef.d.ts b/lib/useEnsuredForwardedRef.d.ts
index 2512358f099351655d309c3dbff1a14bfae945ba..8282846b05f344ac19025d19ddc12ec85ebd782e 100644
--- a/lib/useEnsuredForwardedRef.d.ts
+++ b/lib/useEnsuredForwardedRef.d.ts
@@ -1,3 +1,3 @@
-import { ForwardRefExoticComponent, MutableRefObject, PropsWithoutRef, RefAttributes, RefForwardingComponent } from 'react';
+import { ForwardRefExoticComponent, MutableRefObject, PropsWithoutRef, RefAttributes, ForwardRefRenderFunction } from 'react';
 export default function useEnsuredForwardedRef<T>(forwardedRef: MutableRefObject<T>): MutableRefObject<T>;
-export declare function ensuredForwardRef<T, P = {}>(Component: RefForwardingComponent<T, P>): ForwardRefExoticComponent<PropsWithoutRef<P> & RefAttributes<T>>;
+export declare function ensuredForwardRef<T, P = {}>(Component: ForwardRefRenderFunction<T, P>): ForwardRefExoticComponent<PropsWithoutRef<P> & RefAttributes<T>>;
diff --git a/lib/usePermission.d.ts b/lib/usePermission.d.ts
index d475392903712816e991e3722f548e0e3fd0c36e..7297147693dfa090c1a32f495557998986307e28 100644
--- a/lib/usePermission.d.ts
+++ b/lib/usePermission.d.ts
@@ -7,7 +7,7 @@ interface IMidiPermissionDescriptor extends PermissionDescriptor {
     name: 'midi';
     sysex?: boolean;
 }
-interface IDevicePermissionDescriptor extends PermissionDescriptor {
+interface IDevicePermissionDescriptor {
     name: 'camera' | 'microphone' | 'speaker';
     deviceId?: string;
 }
